File: fixtures/scale/ruby/large/lib/check_modwheel.rb
require_relative 'modwheel_encrypt'
require_relative 'modwheel_decrypt'
require_relative 'corpus_affine'

def check_modwheel
  corpus_affine.each do |text|
    sealed = modwheel_encrypt(text, 14)
    opened = modwheel_decrypt(sealed, 14)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_modwheel.py
from cipherlab.ciphers.modwheel import modwheel_encrypt, modwheel_decrypt
from cipherlab.specs.modwheel_spec import modwheel_spec
from cipherlab.corpus.corpus_affine import corpus_affine


def check_modwheel():
    spec = modwheel_spec()
    for text in corpus_affine():
        sealed = modwheel_encrypt(text, spec["key"])
        opened = modwheel_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "affine"
